retrieve csv without downloading
finding total number of a series(eg. reported cases)
finding the overall death rate, rounded to 2sfs
find total tests conducted if 935310 tests were conducted before databse records
Explain positive rate Find the positive rate
find the number of days new cases(recorded) were more than 100
Demonstrate .count() method and say what it does
Return the actual days where the new cases were more than 1000
Determine the days when the ratio of cases reported to tests conducted is higher than the overall positive_rate.


In [3]:
import pandas as pd

c_data =pd.read_csv(r'https://gist.githubusercontent.com/aakashns/f6a004fa20c84fec53262f9a8bfee775/raw/f309558b1cf5103424cef58e2ecb8704dcd4d74c/italy-covid-daywise.csv')

In [4]:
c_data

date  new_cases  new_deaths  new_tests
0    2019-12-31        0.0         0.0        NaN
1    2020-01-01        0.0         0.0        NaN
2    2020-01-02        0.0         0.0        NaN
3    2020-01-03        0.0         0.0        NaN
4    2020-01-04        0.0         0.0        NaN
..          ...        ...         ...        ...
243  2020-08-30     1444.0         1.0    53541.0
244  2020-08-31     1365.0         4.0    42583.0
245  2020-09-01      996.0         6.0    54395.0
246  2020-09-02      975.0         8.0        NaN
247  2020-09-03     1326.0         6.0        NaN

[248 rows x 4 columns]

In [5]:
#Q: What are the total number of reported cases and deaths related to Covid-19 in Italy?
total_cases = c_data['new_cases'].sum()
total_cases

271515.0

In [6]:
#Q: What is the overall death rate (ratio of reported deaths to reported cases)?
total_deaths = c_data['new_deaths'].sum()
total_deaths

35497.0

In [7]:
print('The number of reported cases is {} whilst the number of reported deaths is {}'.format(total_cases,total_deaths))

The number of reported cases is 271515.0 whilst the number of reported deaths is 35497.0


In [8]:
#Q: What is the overall death rate (ratio of reported deaths to reported cases)?

r = (total_deaths/total_cases)

print(" {:.2f}% is the ratio of reported deaths to reported cases".format(r*100))

 13.07% is the ratio of reported deaths to reported cases


In [10]:
#Q: What is the overall number of tests conducted? A total of 935310 tests were conducted before daily test numbers were reported.

total_initial_tests = 935310

overall_tests = (c_data['new_tests'].sum()) + total_initial_tests 

overall_tests

5214766.0

In [11]:
overall_tests = round(overall_tests, 2)

overall_tests


5214766.0

The positive rate is the percentage of people who test positive for COVID-19 out of the total number of people who are tested. To calculate the positive rate, you can use the following formula:
Positive Rate = (Total Number of Positive Cases) / (Total Number of Tests)

In [12]:
#Find the positive rate
positive_rate = total_cases/overall_tests
positive_rate

0.05206657403227681

In [13]:
#Let's say we want only want to look at the days which had more than 1000 reported cases

c_data['new_cases'] > 1000


0      False
1      False
2      False
3      False
4      False
       ...  
243     True
244     True
245    False
246    False
247     True
Name: new_cases, Length: 248, dtype: bool

In [22]:
w = c_data['new_cases'] 


0         0.0
1         0.0
2         0.0
3         0.0
4         0.0
        ...  
243    1444.0
244    1365.0
245     996.0
246     975.0
247    1326.0
Name: new_cases, Length: 248, dtype: float64

In [28]:
#To find the number of days new cases(recorded) were more than 100
high_days = 0
low_days = 0
case_leap_days = w > 100
for i in case_leap_days:
    if i is True:
        high_days = high_days +1
    else:
        low_days = low_days +1
print('Number of days with cases more than 100 is ', high_days)
print('Number of days with cases less than 100 is ', low_days)

Number of days with cases more than 100 is  188
Number of days with cases less than 100 is  60


In [16]:
c_data.count()

date          248
new_cases     248
new_deaths    248
new_tests     135
dtype: int64

So we can conclude out of 248 instances of recorded cases 188 days had more than 100 cases

In [17]:
#To pull out the data itself, for example i want to pull out data from days were the reported cases was more than 1000

very_high_case_day = c_data[c_data.new_cases > 1000]
very_high_case_day

date  new_cases  new_deaths  new_tests
68   2020-03-08     1247.0        36.0        NaN
69   2020-03-09     1492.0       133.0        NaN
70   2020-03-10     1797.0        98.0        NaN
72   2020-03-12     2313.0       196.0        NaN
73   2020-03-13     2651.0       189.0        NaN
..          ...        ...         ...        ...
241  2020-08-28     1409.0         5.0    65135.0
242  2020-08-29     1460.0         9.0    64294.0
243  2020-08-30     1444.0         1.0    53541.0
244  2020-08-31     1365.0         4.0    42583.0
247  2020-09-03     1326.0         6.0        NaN

[72 rows x 4 columns]

In [18]:
c_data

date  new_cases  new_deaths  new_tests
0    2019-12-31        0.0         0.0        NaN
1    2020-01-01        0.0         0.0        NaN
2    2020-01-02        0.0         0.0        NaN
3    2020-01-03        0.0         0.0        NaN
4    2020-01-04        0.0         0.0        NaN
..          ...        ...         ...        ...
243  2020-08-30     1444.0         1.0    53541.0
244  2020-08-31     1365.0         4.0    42583.0
245  2020-09-01      996.0         6.0    54395.0
246  2020-09-02      975.0         8.0        NaN
247  2020-09-03     1326.0         6.0        NaN

[248 rows x 4 columns]

In [19]:
#Okay babe let's take it a step further...let's try to determine the days when the ratio of cases reported to tests conducted is higher than the overall positive_rate.

v = c_data[c_data['new_cases']/c_data['new_tests']  > positive_rate]
v

date  new_cases  new_deaths  new_tests
111  2020-04-20     3047.0       433.0     7841.0
112  2020-04-21     2256.0       454.0    28095.0
113  2020-04-22     2729.0       534.0    44248.0
114  2020-04-23     3370.0       437.0    37083.0
116  2020-04-25     3021.0       420.0    38676.0
117  2020-04-26     2357.0       415.0    24113.0
118  2020-04-27     2324.0       260.0    26678.0
120  2020-04-29     2091.0       382.0    38589.0
123  2020-05-02     1965.0       269.0    31231.0
124  2020-05-03     1900.0       474.0    27047.0
125  2020-05-04     1389.0       174.0    22999.0
128  2020-05-07     1444.0       369.0    13665.0